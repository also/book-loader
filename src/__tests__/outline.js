function createHtml(str, options={}) {
  const markdownIt = require('markdown-it');
  return markdownIt(options).render(str);
}

it('parses simple outline', () => {
  const outline = require('../outline');
  expect(outline(createHtml(
`
* [One](one.md)
* [Two](two.md)
`
  ))).toMatchSnapshot();
});

it('parses tag in link', () => {
  const outline = require('../outline');
  expect(outline(createHtml(
`
* [*One*](one.md)
* [Two](two.md)
`
  ))).toMatchSnapshot();
});

it('parses link in tag', () => {
  const outline = require('../outline');
  expect(outline(createHtml(
`
* *[One](one.md)*
* [Two](two.md)
`
  ))).toMatchSnapshot();
});

it('parses nested list', () => {
  const outline = require('../outline');
  expect(outline(createHtml(
`
* [One](one.md)
  * [One A](one-a.md)
  * [One b](one-b.md)
* [Two](two.md)
`
  ))).toMatchSnapshot();
});
