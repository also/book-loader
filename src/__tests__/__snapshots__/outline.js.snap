// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignores comments 1`] = `
Object {
  "attrs": Object {},
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "list": false,
          "title": Array [
            Object {
              "attrs": Object {
                "href": "one.md",
              },
              "html": "One",
              "list": false,
              "type": "a",
            },
          ],
          "type": "entry",
        },
        Object {
          "children": Array [],
          "list": false,
          "title": Array [
            Object {
              "attrs": Object {
                "href": "two.md",
              },
              "html": "Two",
              "list": false,
              "type": "a",
            },
          ],
          "type": "entry",
        },
      ],
      "list": true,
      "type": "list",
    },
  ],
  "list": true,
  "simple": false,
  "type": "root",
}
`;

exports[`parses link in tag 1`] = `
Object {
  "attrs": Object {},
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "list": false,
          "title": Array [
            Object {
              "attrs": Object {},
              "children": Array [
                Object {
                  "attrs": Object {
                    "href": "one.md",
                  },
                  "html": "One",
                  "list": false,
                  "type": "a",
                },
              ],
              "list": false,
              "simple": false,
              "type": "em",
            },
          ],
          "type": "entry",
        },
        Object {
          "children": Array [],
          "list": false,
          "title": Array [
            Object {
              "attrs": Object {
                "href": "two.md",
              },
              "html": "Two",
              "list": false,
              "type": "a",
            },
          ],
          "type": "entry",
        },
      ],
      "list": true,
      "type": "list",
    },
  ],
  "list": true,
  "simple": false,
  "type": "root",
}
`;

exports[`parses nested list 1`] = `
Object {
  "attrs": Object {},
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "list": false,
              "title": Array [
                Object {
                  "attrs": Object {
                    "href": "one-a.md",
                  },
                  "html": "One A",
                  "list": false,
                  "type": "a",
                },
              ],
              "type": "entry",
            },
            Object {
              "children": Array [],
              "list": false,
              "title": Array [
                Object {
                  "attrs": Object {
                    "href": "one-b.md",
                  },
                  "html": "One b",
                  "list": false,
                  "type": "a",
                },
              ],
              "type": "entry",
            },
          ],
          "list": true,
          "title": Array [
            Object {
              "attrs": Object {
                "href": "one.md",
              },
              "html": "One",
              "list": false,
              "type": "a",
            },
          ],
          "type": "entry",
        },
        Object {
          "children": Array [],
          "list": false,
          "title": Array [
            Object {
              "attrs": Object {
                "href": "two.md",
              },
              "html": "Two",
              "list": false,
              "type": "a",
            },
          ],
          "type": "entry",
        },
      ],
      "list": true,
      "type": "list",
    },
  ],
  "list": true,
  "simple": false,
  "type": "root",
}
`;

exports[`parses simple outline 1`] = `
Object {
  "attrs": Object {},
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "list": false,
          "title": Array [
            Object {
              "attrs": Object {
                "href": "one.md",
              },
              "html": "One",
              "list": false,
              "type": "a",
            },
          ],
          "type": "entry",
        },
        Object {
          "children": Array [],
          "list": false,
          "title": Array [
            Object {
              "attrs": Object {
                "href": "two.md",
              },
              "html": "Two",
              "list": false,
              "type": "a",
            },
          ],
          "type": "entry",
        },
      ],
      "list": true,
      "type": "list",
    },
  ],
  "list": true,
  "simple": false,
  "type": "root",
}
`;

exports[`parses tag in link 1`] = `
Object {
  "attrs": Object {},
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "list": false,
          "title": Array [
            Object {
              "attrs": Object {
                "href": "one.md",
              },
              "html": "<em>One</em>",
              "list": false,
              "type": "a",
            },
          ],
          "type": "entry",
        },
        Object {
          "children": Array [],
          "list": false,
          "title": Array [
            Object {
              "attrs": Object {
                "href": "two.md",
              },
              "html": "Two",
              "list": false,
              "type": "a",
            },
          ],
          "type": "entry",
        },
      ],
      "list": true,
      "type": "list",
    },
  ],
  "list": true,
  "simple": false,
  "type": "root",
}
`;
